cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -490,6 +503,44 @@ func expandKubeMasterURL() (string, error) {
 	return parsedURL.String(), nil
 }
 
+func (c *etcdConfig) newClient() (client *etcd.Client, err error) {
```

Comment:
if possible, I'd like to see this converted to the new etcd client to match the apiserver (the old client's construction is odd if you have a CA but no cert/key, or vice-versa)